src/app/iteration/iteration.service.ts

```diff
@@ -80,10 +81,35 @@ export class IterationService {
   }
 
   /**
-   * Get locally saved iterations' reference
+   * Update an existing iteration
+   * @param iteration - Updated iteration
+   * @return updated iteration's reference from the list
    */
-  getIterationList() {
-    return this.iterations;
+  updateIteration(iteration: IterationModel): Promise<IterationModel> {
+    return this.http
+      .patch(iteration.links.self, { headers: this.headers })
+      .toPromise()
+      .then (response => {
+        if (/^[5, 4][0-9]/.test(response.status.toString())) {
```

Comment:
Wouldn't it be just better to check if `response.status >= 400 && response.status <= 599`? regex is not for free I guess. On top of that this code is duplicated above - maybe small extract?

Don't we have some meaningful explanation of what was the error in the `response`? If so maybe it's good to populate it as a part of `new Error(....)`?